zokrates_core/src/flatten/mod.rs

```diff
@@ -2016,21 +2016,58 @@ impl<'ast, T: Field> Flattener<'ast, T> {
                 }
             }
             ZirStatement::Assertion(e) => {
-                // naive approach: flatten the boolean to a single field element and constrain it to 1
+                match e {
+                    BooleanExpression::And(..) => {
+                        let mut iter = self.unwrap_and_chain(e).into_iter();
```

Comment:
for boolean in self.unwrap_and_chain(e) {
    self.flatten_statement(
          symbols,
          statements_flattened,
          ZirStatement::Assertion(boolean),
      )
}